synapse/cortex.py

```diff
@@ -580,6 +580,95 @@ def _onFeedTagDel(self, snap, mesg):
     # def _addSynUndo(self, snap, items):
         # TODO apply splices in reverse
 
+    def _addSynIngest(self, snap, items):
+        # Check and wrap items in a list in the event that
+        # a single ingest definition was passed in.
+        if isinstance(items, dict):
+            items = [items]
+
+        for item in items:
+            try:
+                pnodes = list(self._genSynIngestNodes(item))
```

Comment:
Any reason we're not actually making this a generator and calling snap.addNode() for each, then yielding?